src/App.jsx

```diff
@@ -1,11 +1,31 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import './App.css'
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+// custom hook to store todos after reloading page
+function useSemiPersistentState() {
+    // read saved todo list from localStorage
+    const savedTodoList = localStorage.getItem('savedTodoList');
```

Comment:
you could put all logic into useState here:
 const [todoList, setTodoList] = useState(
    JSON.parse(localStorage.getItem("savedTodoList")) || []
  );